### ISSUE TITLE
The inbound traffic statistics of the LAN interface are incorrect.

### ISSUE THREAD

**Issue by dongc123** [openwrt/openwrt#13457]
### Describe the bug

**The inbound traffic statistics of the LAN interface are incorrect.**
![微信截图_20230915160331](https://github.com/openwrt/openwrt/assets/109882914/348f8f13-3d51-469f-822a-bda1da1c0e18)

### OpenWrt version

OpenWrt 23.05.0-rc3 r23389-5deed175a5

### OpenWrt target/subtarget

x86/64

### Device

x86/64

### Image kind

Official downloaded image

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by brada4** [comment]
How is it incorrect? It is quite credible packets and megabytes numbers?

**Comment by dongc123** [comment]
> 怎么会不正确呢？这是相当可信的数据包和兆字节数字吗？

![微信截图_20230915195744](https://github.com/openwrt/openwrt/assets/109882914/7803b6be-a609-43dd-b2cd-641d13bc3fad)
![微信截图_20230915195924](https://github.com/openwrt/openwrt/assets/109882914/b43929d1-ddd8-4939-860d-babda47caf25)
The lan bridge has three network ports eth0 eth1 eht2. The inbound traffic statistics of eth2 are 26GB. The lan bridge shows that the inbound traffic is 91MB.

**Comment by brada4** [comment]
What is in /sys/class/net/br-lan/statistics/{tx|rx}_{bytes|packets} ?

**Comment by dongc123** [comment]
> /sys/class/net/br-lan/statistics/{tx|rx}_{bytes|packets} 中有什么？

![image](https://github.com/openwrt/openwrt/assets/109882914/11bf56f8-72b9-4aa0-b68e-74bad9a2f796)

**Comment by brada4** [comment]
Luci reflects what kernel accounted?

**Comment by dongc123** [comment]
Are you talking about a luci problem?

**Comment by KanjiMonster** [comment]
Just an educated guess, but the rx/tx counters of `lan` is the amount of traffic sent to the bridge interface `br-lan`, so the OpenWrt router itself, and not the aggregated traffic over all bridge ports.

So packets that just get forwarded for example between `eth2` and `eth3` do not count. But if you access luci, then the request and the reply get counted.

**Comment by brada4** [comment]
@KanjiMonster you are absolutely correct, the counters are for  bridge's own MAC+IP, excluding forwarded traffic.
If one disables all aggregating offloads in all network cards input/output sums match.

**Comment by dongc123** [comment]
> 只是一个有根据的猜测，但 rx/tx 计数器`lan`是发送到桥接接口的流量量`br-lan`，因此是 OpenWrt 路由器本身，而不是所有桥接端口上的聚合流量。
> 
> `eth2`因此，仅在和之间转发的数据包`eth3`不计入在内。但如果您访问 luci，则请求和回复都会被计数。

I remember that the OpenWrt 23.05.0-rc2 version lan bridge tx rx statistics are correct

**Comment by dongc123** [comment]
> @KanjiMonster你是完全正确的，计数器是针对网桥自己的MAC+IP，不包括转发的流量。 如果禁用所有网卡输入/输出总和匹配中的所有聚合卸载。

Then why are the statistics sent from the wan port and the inbound lan correct?

**Comment by brada4** [comment]
If packet is offloaded in firewall then it is transported from one interface with cable to another, no intermediary counters involved.
If interface supports some bulk offloads like gro tso etc then 1 packet may match multiple physical frames.

**Comment by KanjiMonster** [comment]
> > @KanjiMonster你是完全正确的，计数器是针对网桥自己的MAC+IP，不包括转发的流量。 如果禁用所有网卡输入/输出总和匹配中的所有聚合卸载。
> 
> Then why are the statistics sent from the wan port and the inbound lan correct?

I don't fully understand the question, likely the answer is layer 2 forwarding versus layer 3 routing.

So on br-lan between eth2 and eth3 the packets go directly between both ports and the br-lan interface in the kernel, and are never sent to the br-lan interface.

But for wan -> eth3 the packets are first received at the wan port, then the src and dst macs get changed to br-lan's mac and eth3's mac, and then the packet gets sent from br-lan to eth3.

So basically this is wan -> br-lan -> eth3.

**Comment by KanjiMonster** [comment]
> > But for ... wan ...
> 
> ... nftables offload bypasses all that br-lan rituals.

But the question was why are they correct, not incorrect, so I assumed no offloading shenanigans involved there.

And obviously offloading will screw up counters in many ways. Which is why AFAICT there is support for explicit offloaded stats.

**Comment by brada4** [comment]
`bridge vlan show -s` 
Shows some incrementing numbers with physical adapters and zeroes with dsa (no DSA in OP)

**Comment by LGA1150** [comment]
Two reasons:

1. A bridge device does not account packets forwarded to another bridge port.
2. A bridge device is a virtual interface, which can be set up and torn down at runtime. When this happens, the software counter will be reset.
